src/components/common/DeactivatableLink.tsx

```diff
@@ -0,0 +1,37 @@
+import { FunctionComponent } from "react";
+import Link, { LinkProps } from "next/link";
+import { classNames } from "@/utils/helpers";
+
+export type DeactivatableLinkProps = {
+	disabled?: boolean;
+};
+
+/**
+ * A link that can be disabled. This is required because Next.js does not support the 'disabled' attribute on Links by default.
+ */
+const DeactivatableLink: FunctionComponent<
+	Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, keyof LinkProps> &
+		LinkProps &
+		DeactivatableLinkProps
```

Comment:
`LinkProps` is not being removed (omitted) - keys (of `AnchorHTMLAttributes<HTMLAnchorElement>` that match LinkProps are being removed. This is so that they don't overlap. I don't actually know if it's a type error if they overlap, but I assumed it was as this is how NextJS's `Link` is developed.

https://github.com/vercel/next.js/blob/3b8bbe197fcf4059f715bb3d2666f0a018878e84/packages/next/src/client/link.tsx#L246

Reply:
Hm, interesting. I see now that the reason for omitting those keys from `AnchorHTMLAttributes<HTMLAnchorElement>` and then "adding them back" isn't really "adding them back", but rather creating an intersection type. Not sure why it's important to have an intersection type here, but I now understand the implementation itself.